docs(concepts): shared modules allow multiple versions in a single build (#3780)
webpack/webpack.js.org: src/content/concepts/module-federation.md

## Concept goals

- It should be possible to expose and use any module type that webpack supports.
- Chunk loading should load everything needed in parallel (web: single round-trip to server).
- Control from consumer to container
    - Overriding modules is a one-directional operation.
    - Sibling containers cannot override each other's modules.
- Concept should be environment-independent.
    - Usable in web, Node.js, etc.

> - Relative and absolute request in shared.

Reviewer: ```suggestion
- Relative and absolute request in shared:
```

    - Will always be provided, even if not used.
    - Will resolve relative to `config.context`.
    - Do not use a `requiredVersion` by default.
- Module requests in shared
    - are only provided when they are used.
    - will match all used equal module requests in your build.
    - will provide all matching modules.
    - will extract `requiredVersion` from package.json at this position in the graph.

>     - could provide and consume multiple different version when you have nested node_modules.

Reviewer: Why isnt this capitalized? Given the previous list...

> - Module requests with trailing / in shared
>     - will match all module requests with this prefix.

Reviewer: ```suggestion
- Module requests with trailing `/` in shared will match all module requests with this prefix.
```

> - New packageName option to choose package name inorder to look for a `requiredVersion`. For module requests it is automatically inferred by default.

Reviewer: I dont see how this belongs to this list, this new option, where does it live? How do you use it? Is it really a part of the concept?
```suggestion
- New `packageName` option allows setting a package name to look for a `requiredVersion`. It is automatically inferred for the module requests by default.
```

Author: https://github.com/webpack/webpack/blob/c00fec3aa7c79476e5b61651558f5804a48b3bac/schemas/plugins/container/ModuleFederationPlugin.json#L367

> - `requiredVersion` can now be `false` when automatic infer should be disabled.

Reviewer: This does not belong to concept goals list for sure

Author: Where do you think the respective information would be a best fit?

### Components library as container

Many applications share a common components library which could be built as a container with each component exposed. Each application consumes components from the components library container. Changes to the components library can be separately deployed without the need to re-deploy all applications. The application automatically uses the up-to-date version of the components library.

> ### Further reading

Reviewer: Its done via frontmatter at the top of md file
